### ISSUE TITLE
模型蒸馏压缩问题

### ISSUE THREAD

**Issue by raymondchu-z** [PaddlePaddle/PaddleDetection#3723]
按照文档进行训练和评估，蒸馏后的模型精度(13%)大幅低于原单独训练的同网络模型(68%)。怀疑是slim_config的问题，但是没有文档说明怎么改。求助

按照文档，训练和评估的命令：使用单独训练过的mAP为68%的模型权重作为预训练权重，训练好的PPYOLOv2_r50vd作为教师模型
~~~
python tools/train.py -c configs/my_config/ppyolo_r18vd.yml --slim_config configs/slim/my_slim/ppyolo_r18_distill_prune_0716.yml -o output/ppyolo_r18vd_1/model_final.pdparams

python -u tools/eval.py  -c configs/my_config/ppyolo_r18vd0716.yml --slim_config configs/slim/my_slim/ppyolo_r18_distill_prune_0716.yml -o weights=output/ppyolo_r18/model_final.pdparams --classwise
~~~
训练过程如下。看loss还挺正常的，因为之前训练PPYOLOv2精度89%，loss也是30+
> [07/17 13:00:03] ppdet.engine INFO: Epoch: [4] [16420/16496] learning_rate: 0.000000 loss_xy: 0.587560 loss_wh: 0.365152 loss_iou: 2.711400 loss_obj: 3.826950 loss_cls: 0.861535 loss: 19.428253 distill_loss: 8.393956 teacher_loss: 43.637764 eta: 0:01:03 batch_cost: 0.6250 data_cost: 0.2446 ips: 19.1997 images/s
[07/17 13:00:16] ppdet.engine INFO: Epoch: [4] [16440/16496] learning_rate: 0.000000 loss_xy: 0.640538 loss_wh: 0.422117 loss_iou: 2.862279 loss_obj: 4.935560 loss_cls: 0.790083 loss: 18.715614 distill_loss: 9.225327 teacher_loss: 43.467583 eta: 0:00:46 batch_cost: 0.5673 data_cost: 0.2132 ips: 21.1525 images/s
[07/17 13:00:32] ppdet.engine INFO: Epoch: [4] [16460/16496] learning_rate: 0.000000 loss_xy: 0.625355 loss_wh: 0.402897 loss_iou: 2.825300 loss_obj: 4.470171 loss_cls: 0.749340 loss: 17.097496 distill_loss: 7.658364 teacher_loss: 44.041996 eta: 0:00:29 batch_cost: 0.6983 data_cost: 0.3151 ips: 17.1842 images/s
[07/17 13:00:44] ppdet.engine INFO: Epoch: [4] [16480/16496] learning_rate: 0.000000 loss_xy: 0.516497 loss_wh: 0.332761 loss_iou: 2.505523 loss_obj: 3.681858 loss_cls: 0.645647 loss: 16.985310 distill_loss: 8.750958 teacher_loss: 48.006203 eta: 0:00:13 batch_cost: 0.4775 data_cost: 0.1240 ips: 25.1322 images/s
[07/17 13:00:55] ppdet.utils.checkpoint INFO: Save checkpoint: output/ppyolo_r18_distill_prune_0716

但是评估结果只有13%，这就不正常了，单独训练的都有68%。

压缩的配置。猜测是这里出问题，但是不知道怎么改。样例中没有对r18这个backbone来蒸馏的。
~~~
_BASE_: [
  '/home/zlm/9.github/PaddleDetection/configs/my_config/mdc_ppyolo/ppyolov2_r50vd_mdc_5e_4gpu_resample_0713_2.yml',
]

pretrain_weights: output/ppyolov2_r50vd_mdc_5e_4gpu_resample_0713_2/model_final.pdparams

slim: DistillPrune

distill_loss: DistillYOLOv3Loss

DistillYOLOv3Loss:
  weight: 1000

pruner: Pruner

Pruner:
  criterion: l1_norm
  pruned_params: ['conv2d_27.w_0', 'conv2d_28.w_0', 'conv2d_29.w_0',
                  'conv2d_30.w_0', 'conv2d_31.w_0', 'conv2d_32.w_0',
                  'conv2d_34.w_0', 'conv2d_35.w_0', 'conv2d_36.w_0',
                  'conv2d_37.w_0', 'conv2d_38.w_0', 'conv2d_39.w_0',
                  'conv2d_41.w_0', 'conv2d_42.w_0', 'conv2d_43.w_0',
                  'conv2d_44.w_0', 'conv2d_45.w_0', 'conv2d_46.w_0']
  pruned_ratios: [0.5,0.5,0.5,0.5,0.5,0.5,0.7,0.7,0.7,0.7,0.7,0.7,0.8,0.8,0.8,0.8,0.8,0.8]
~~~

**Comment by yghstill** [comment]
@raymondchu-z 蒸馏加剪裁一般加载当前要剪裁的模型的训好的pretrain_weights，你再试下加载ppyolo_r18vd的模型再训下看看。如果只蒸馏，可以加载ppyolov2_r50vd的模型
